### PR TITLE
Blocks and Items

### PR DESCRIPTION
**Pull request by OroArmor**
Still need to work my way through the sculk gen logic

### CODE REVIEW

File: mappings/net/minecraft/block/AbstractLichenBlock.mapping [hunk 1]
```diff
@@ -31,14 +31,30 @@ CLASS net/minecraft/unmapped/C_qpwigkki net/minecraft/block/AbstractLichenBlock
...
+	METHOD m_pqscucoi unpackDirections (B)Ljava/util/Set;
+		ARG 0 packed
 	METHOD m_sbqgnavz canGrowOn (Lnet/minecraft/unmapped/C_peaveboq;Lnet/minecraft/unmapped/C_xpuuihxf;Lnet/minecraft/unmapped/C_hynzadkk;Lnet/minecraft/unmapped/C_txtbiemp;)Z
 		ARG 0 world
 		ARG 1 direction
 		ARG 2 pos
 		ARG 3 state
 	METHOD m_scdkefrx getProperty (Lnet/minecraft/unmapped/C_xpuuihxf;)Lnet/minecraft/unmapped/C_xhwijdsd;
 		ARG 0 direction
+	METHOD m_uoihoidn getLichenGrowThing ()Lnet/minecraft/unmapped/C_twpgwpwn;
```
Comment (NoComment1105) [inline mappings/net/minecraft/block/AbstractLichenBlock.mapping]: GrowThing?

File: mappings/net/minecraft/block/C_eprlufrn.mapping [hunk 2]
```diff
@@ -0,0 +1 @@
+CLASS net/minecraft/unmapped/C_eprlufrn net/minecraft/block/C_eprlufrn
```
Comment (NoComment1105) [inline mappings/net/minecraft/block/C_eprlufrn.mapping]: Why the empty un-named class?

File: mappings/net/minecraft/block/C_uevszakc.mapping [hunk 3]
```diff
@@ -0,0 +1,5 @@
+CLASS net/minecraft/unmapped/C_uevszakc net/minecraft/block/C_uevszakc
```
Comment (NoComment1105) [inline mappings/net/minecraft/block/C_uevszakc.mapping]: Is the hashed name intentional?

**Review by NoComment1105 — CHANGES_REQUESTED** [review]
Looking good so far, a couple of comments

File: mappings/net/minecraft/block/AbstractLichenBlock.mapping (hunk 1 above)
Comment (OroArmor) [reply mappings/net/minecraft/block/AbstractLichenBlock.mapping]: Wip name. As for the rest

File: mappings/net/minecraft/block/AbstractLichenBlock.mapping [hunk 4]
```diff
@@ -31,14 +32,30 @@ CLASS net/minecraft/unmapped/C_qpwigkki net/minecraft/block/AbstractLichenBlock
 		ARG 1 direction
 	METHOD m_jiwfgtpt (Ljava/util/EnumMap;)V
 		ARG 0 shapes
+	METHOD m_khvcoyem openFaces (Lnet/minecraft/unmapped/C_txtbiemp;)Ljava/util/Set;
```
Comment (IotaBread) [inline mappings/net/minecraft/block/AbstractLichenBlock.mapping]:
```suggestion
	METHOD m_khvcoyem getOpenFaces (Lnet/minecraft/unmapped/C_txtbiemp;)Ljava/util/Set;
```

File: mappings/net/minecraft/block/ExperienceBlock.mapping [hunk 5]
```diff
@@ -0,0 +1,5 @@
+CLASS net/minecraft/unmapped/C_nrkgonet net/minecraft/block/ExperienceBlock
```
Comment (IotaBread) [inline mappings/net/minecraft/block/ExperienceBlock.mapping]: This sounds like the block is made out of experience

File: mappings/net/minecraft/block/GlowLichenBlock.mapping [hunk 6]
```diff
@@ -1,4 +1,5 @@
 CLASS net/minecraft/unmapped/C_anxwrrtr net/minecraft/block/GlowLichenBlock
+	FIELD f_fjlfksey lichenGrow Lnet/minecraft/unmapped/C_twpgwpwn;
```
Comment (IotaBread) [inline mappings/net/minecraft/block/GlowLichenBlock.mapping]:
```suggestion
	FIELD f_fjlfksey spreadBehavior Lnet/minecraft/unmapped/C_twpgwpwn;
```

File: mappings/net/minecraft/block/MangroveLeavesBlock.mapping [hunk 7]
```diff
@@ -0,0 +1,5 @@
+CLASS net/minecraft/unmapped/C_rygjfpse net/minecraft/block/MangroveLeavesBlock
+	FIELD f_pwwsrwlg GROW_CHANGE I
+	METHOD m_iojhiogk closeToOtherPropagule (Lnet/minecraft/unmapped/C_vdvbsyle;Lnet/minecraft/unmapped/C_hynzadkk;)Z
```
Comment (IotaBread) [inline mappings/net/minecraft/block/MangroveLeavesBlock.mapping]:
```suggestion
	METHOD m_iojhiogk isCloseToOtherPropagule (Lnet/minecraft/unmapped/C_vdvbsyle;Lnet/minecraft/unmapped/C_hynzadkk;)Z
```

File: mappings/net/minecraft/block/LichenSpreadBehavior.mapping [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+	METHOD m_jfrwlpgd (Lnet/minecraft/unmapped/C_txtbiemp;Lnet/minecraft/unmapped/C_xpuuihxf;)Z
+		ARG 2 dir
+	METHOD m_lajlmezu spreadFromFacingToDirection (Lnet/minecraft/unmapped/C_txtbiemp;Lnet/minecraft/unmapped/C_vdvbsyle;Lnet/minecraft/unmapped/C_hynzadkk;Lnet/minecraft/unmapped/C_xpuuihxf;Lnet/minecraft/unmapped/C_xpuuihxf;Z)Ljava/util/Optional;
+		ARG 1 state
+		ARG 2 world
+		ARG 3 pos
+		ARG 4 facing
+		ARG 5 direction
+		ARG 6 postProcess
+	METHOD m_mamsfnxt spreadable (Lnet/minecraft/unmapped/C_vdvbsyle;Lnet/minecraft/unmapped/C_twpgwpwn$C_evcfujfb;Z)Ljava/util/Optional;
```
Comment (IotaBread) [inline mappings/net/minecraft/block/LichenSpreadBehavior.mapping]:
```suggestion
	METHOD m_mamsfnxt isSpreadable (Lnet/minecraft/unmapped/C_vdvbsyle;Lnet/minecraft/unmapped/C_twpgwpwn$C_evcfujfb;Z)Ljava/util/Optional;
```
?

File: mappings/net/minecraft/block/SculkVeinBlock.mapping [hunk 9]
```diff
@@ -0,0 +1,23 @@
+CLASS net/minecraft/unmapped/C_bjvwjobf net/minecraft/block/SculkVeinBlock
+	FIELD f_lhdhlpuz samePosLichenGrowBehavior Lnet/minecraft/unmapped/C_twpgwpwn;
```
Comment (IotaBread) [inline mappings/net/minecraft/block/SculkVeinBlock.mapping]:
```suggestion
	FIELD f_lhdhlpuz samePosLichenSpreadBehavior Lnet/minecraft/unmapped/C_twpgwpwn;
```

File: mappings/net/minecraft/block/SculkVeinBlock.mapping [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+	METHOD m_jusulnlz spreadInFirstValidDirection (Lnet/minecraft/unmapped/C_vdvbsyle;Lnet/minecraft/unmapped/C_hynzadkk;Lnet/minecraft/unmapped/C_txtbiemp;Ljava/util/Collection;)Z
+		ARG 0 world
+		ARG 1 pos
+		ARG 2 state
+		ARG 3 directions
+	METHOD m_lmxweihp canSpread (Lnet/minecraft/unmapped/C_vdvbsyle;Lnet/minecraft/unmapped/C_txtbiemp;Lnet/minecraft/unmapped/C_hynzadkk;)Z
+		ARG 0 world
+		ARG 1 state
+		ARG 2 pos
+	METHOD m_pkzupcww getSamePosLichenGrowBehavior ()Lnet/minecraft/unmapped/C_twpgwpwn;
```
Comment (IotaBread) [inline mappings/net/minecraft/block/SculkVeinBlock.mapping]:
```suggestion
	METHOD m_pkzupcww getSamePosLichenSpreadBehavior ()Lnet/minecraft/unmapped/C_twpgwpwn;
```

File: mappings/net/minecraft/block/sculk/SculkBehavior.mapping [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+		METHOD m_ljlofwsf (Lnet/minecraft/unmapped/C_hynzadkk;)Z
+			ARG 0 pos
+		METHOD m_obzbqurs (Lnet/minecraft/unmapped/C_eprlufrn$C_gzpsastu;)Ljava/lang/Integer;
+			ARG 0 arg
+		METHOD m_qhrumuao shouldUpdate (Lnet/minecraft/unmapped/C_vdvbsyle;Lnet/minecraft/unmapped/C_hynzadkk;Z)Z
+			ARG 1 world
+			ARG 2 pos
+			ARG 3 forceUpdate
+		METHOD m_qjqwrgss getDirections ()Ljava/util/Set;
+		METHOD m_tvkgrthy shuffledDirections (Ljava/util/Random;)Ljava/util/List;
```
Comment (IotaBread) [inline mappings/net/minecraft/block/sculk/SculkBehavior.mapping]:
```suggestion
		METHOD m_tvkgrthy getShuffledDirections (Ljava/util/Random;)Ljava/util/List;
```

File: mappings/net/minecraft/item/ItemStack.mapping [hunk 12]
```diff
@@ -297,6 +299,8 @@ CLASS net/minecraft/unmapped/C_sddaxwyk net/minecraft/item/ItemStack
 		ARG 1 right
 	METHOD m_sxojkkcd setCustomName (Lnet/minecraft/unmapped/C_rdaqiwdt;)Lnet/minecraft/unmapped/C_sddaxwyk;
 		ARG 1 name
+	METHOD m_sxworkis is (Lnet/minecraft/unmapped/C_cjzoxshv;)Z
```
Comment (IotaBread) [inline mappings/net/minecraft/item/ItemStack.mapping]: How about `matches`?

**Review by NoComment1105 — APPROVED** [review]

**Pull request merged**